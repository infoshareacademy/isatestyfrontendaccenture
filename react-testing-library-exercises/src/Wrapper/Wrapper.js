export const Wrapper = ({ children }) => (
    <div style={{
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center'
    }}>
        {children}
    </div>
)
