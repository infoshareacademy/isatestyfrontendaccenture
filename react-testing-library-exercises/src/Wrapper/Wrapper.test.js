// Napisz podane testy!
import { render, screen } from '@testing-library/react';
import { Wrapper } from './Wrapper';


describe('Wrapper', () => {
    beforeEach(() => expect.hasAssertions());

    test('should create div with correct styles', () => {
        const { container } = render(<Wrapper />);

        expect(container.firstChild).toMatchSnapshot();
    });

    test('should show passed children', () => {
        const Child = () => <div>Child</div>;
        render(
            <Wrapper>
                <Child />
            </Wrapper>
        );

        expect(screen.getByText('Child')).toBeInTheDocument();
    });
});
