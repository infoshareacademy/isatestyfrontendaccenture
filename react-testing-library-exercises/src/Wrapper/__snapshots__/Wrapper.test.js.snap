// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapper should create div with correct styles 1`] = `
<div
  style="display: flex; flex-direction: column; align-items: center;"
/>
`;
